### PR TITLE
Mc 31 implement music and soundeffects

### PR DESCRIPTION
**Pull request by danielrittrich**

### CODE REVIEW

File: Merge Conflict/Assets/Scripts/AudioManager/AudioManager.cs [hunk 1]
```diff
@@ -0,0 +1,218 @@
+/**********************************************************************************************************************
+Name:          AudioManager
+Description:   Manages all sounds and background music that will be played.
```
Comment (Xipit) [inline Merge Conflict/Assets/Scripts/AudioManager/AudioManager.cs:3]: Pls mention that the methods are to be used by other scripts and that the background music is handled by the AudioManager itself

File: Merge Conflict/Assets/Scripts/AudioManager/AudioManager.cs [hunk 2]
```diff
@@ -0,0 +1,218 @@
...
+    public float footstepInterval = 0.18f;
+
+    // ?  Maybe other sounds or environment sounds 
+    /*    (conveyorbelt, component walking, component dragging, component dropping, individual sounds for upgrades, 
+           open menu, close menu, component returning to desk, trash walkin on desk, level up, close menu, quest completed,
+           get component when throwing one away (upgrade), buy component, selling wrong component,  ... )  */
+
+    // TODO: don't play dropComponent Sound when selling a component or throwing it in the trash can
+
+    private Coroutine playNextSongCoroutine;
```
Comment (Xipit) [inline Merge Conflict/Assets/Scripts/AudioManager/AudioManager.cs:77]: make it more clear that "nextSong" refers to the background music. This includes the rest of the "song" methods.

File: Merge Conflict/Assets/Scripts/AudioManager/AudioManager.cs [hunk 3]
```diff
@@ -0,0 +1,218 @@
...
+    {
+        dropComponentAudioSource.PlayOneShot(dropComponentSound);
+    }
+
+    public void PlayPickUpComponentSound()
+    {
+        pickUpComponentAudioSource.PlayOneShot(pickUpComponentSound);
+    }
+
+    public void StartFootstepLoop()
```
Comment (Xipit) [inline Merge Conflict/Assets/Scripts/AudioManager/AudioManager.cs:192]: make it more clear that the component footstep is meant

File: Merge Conflict/Assets/Scripts/AudioManager/AudioMixerLevels.cs [hunk 4]
```diff
@@ -0,0 +1,63 @@
+/**********************************************************************************************************************
+Name:          AudioMixerLevels
+Description:   Handles the volume level of the slider in the audio settings and saves the values in the playerprefs.
```
Comment (Xipit) [inline Merge Conflict/Assets/Scripts/AudioManager/AudioMixerLevels.cs:3]: mention that the script is attached directly to the slider and is not singular manager, such as the AudioManager.

File: Merge Conflict/Assets/Scripts/AudioManager/AudioMixerLevels.cs [hunk 5]
```diff
@@ -0,0 +1,63 @@
...
+using UnityEngine;
+using UnityEngine.UI;
+using UnityEngine.Audio;
+
+public class AudioMixerLevels : MonoBehaviour
+{
+    public Slider thisSlider;
+    public AudioMixer audioMixer;
+
+    [Tooltip("Must be 'volumeEffects' or 'volumeMusic' !")]
```
Comment (Xipit) [inline Merge Conflict/Assets/Scripts/AudioManager/AudioMixerLevels.cs:19]: refactor exposedParameter and playerPrefsKey to be driven by an enum. 
In places where these variables are used, should be a function call, that takes in the enum as parameter and returns the approriate string ("volumeEffects", ...)

Reason is that it is to easy to accidentily change these values in the editor and not notice the error.

File: Merge Conflict/Assets/Scripts/ComponentMovement.cs [hunk 6]
```diff
@@ -373,4 +388,12 @@ public bool GetIsReturningToDesk()
     {
         return _isReturningToDesk;
     }
+
+    private IEnumerator SetIsNotWalking()
```
Comment (Xipit) [inline Merge Conflict/Assets/Scripts/ComponentMovement.cs:392]: mention delay in Coroutine name

**Review by Xipit — COMMENTED** [review]
The sound effects and music really bring the game to life 👍 

General architecture is good, only some minor nitpicks about foolproofing and naming.

If you addressed my comments, you can merge on your own

File: Merge Conflict/Assets/Scripts/AudioManager/AudioManager.cs (hunk 1 above)
Comment (danielrittrich) [reply Merge Conflict/Assets/Scripts/AudioManager/AudioManager.cs:3]: done

File: Merge Conflict/Assets/Scripts/AudioManager/AudioManager.cs (hunk 3 above)
Comment (danielrittrich) [reply Merge Conflict/Assets/Scripts/AudioManager/AudioManager.cs:192]: done

File: Merge Conflict/Assets/Scripts/AudioManager/AudioMixerLevels.cs (hunk 4 above)
Comment (danielrittrich) [reply Merge Conflict/Assets/Scripts/AudioManager/AudioMixerLevels.cs:3]: done

**Pull request merged**